src/solvers/interior_point_Newton.jl

```diff
@@ -0,0 +1,385 @@
+
+```math
+\begin{aligned}
+\min_{p ∈\mathcal{M}} &f(p)\\
+\text{subject to } &g_i(p)\leq 0 \quad \text{ for } i= 1, …, m,\\
+\quad &h_j(p)=0 \quad \text{ for } j=1,…,n,
+\end{aligned}
+```
+
+This algorithms iteratively solves the linear system based on extending the KKT system
+by a slack variable `s`.
+
+```math
+\operatorname{J} F(p, μ, λ, s)[X, Y, Z, W] = -F(p, μ, λ, s),
+\text{ where }
+X ∈ T_p\mathcal M, Y,W ∈ ℝ^m, Z ∈ ℝ^n,
+```
+see [`CondensedKKTVectorFieldJacobian`](@ref) and [`CondensedKKTVectorField`](@ref), respectively,
+for the reduced form, this is usually solved in.
+From the resulting `X` and `Z` in the reeuced form, the other two can be computed.
+
+Note that since the vector field ``F`` includes the gradients of the constraint
+functions ``g,h`, its gradient or Jacobian requires the Hessians of the constraints.
+
+For that seach direction a line search is performed, that additionally ensures that
+the constraints are further fulfilled.
+
+# Input
+
+* `M`:      a manifold ``\mathcal M``
+* `f`:      a cost function ``f : \mathcal M → ℝ`` to minimize
+* `grad_f`: the gradient ``\operatorname{grad} f : \mathcal M → T \mathcal M`` of ``f``
+* `Hess_f`: the Hessian ``\operatorname{Hess}f(p): T_p\mathcal M → T_p\mathcal M``, ``X ↦ \operatorname{Hess}F(p)[X] = ∇_X\operatorname{grad}f(p)``
+* `p=`[`rand`](@extref Base.rand-Tuple{AbstractManifold})`(M)`: an initial value ``p  ∈  \mathcal M``
+
+# Keyword arguments
+
+* `equality_constraints=nothing`: the number ``n`` of equality constraints.
+* `g=nothing`: the inequality constraints
+* `grad_g=nothing`: the gradient of the inequality constraints
+* `grad_h=nothing`: the gradient of the equality constraints
+* `gradient_range=nothing`: specify how gradients are represented, where `nothing` is equivalent to [`NestedPowerRepresentation`](@extref)
+* `gradient_equality_range=gradient_range`: specify how the gradients of the equality constraints are represented
+* `gradient_inequality_range=gradient_range`: specify how the gradients of the inequality constraints are represented
+* `h=nothing`: the equality constraints
+* `Hess_g=nothing`: the Hessian of the inequality constraints
+* `Hess_h=nothing`: the Hessian of the equality constraints
+* `inequality_constraints=nothing`: the number ``m`` of inequality constraints.
+* `λ=ones(size(h(M,x),1))`: the Lagrange multiplier with respect to the equality constraints ``h``
+* `μ=ones(size(h(M,x),1))`: the Lagrange multiplier with respect to the inequality constraints ``g``
+* `s=μ`: initial value for the slack variables
+* `σ=μ's/length(μ)`: ? (TODO find details about barrier parameter)
+* `stopping_criterion::StoppingCriterion=`[`StopAfterIteration`](@ref)`(200)`[` | `](@ref StopWhenAny)[`StopWhenChangeLess`](@ref)`(1e-5)`: a stopping criterion
+* `retraction_method=[`default_retraction_method`](@extref)`(M, typeof(p))`: the retraction to use, defaults to the default set `M` with respect to the representation for `p` chosen.
```

Comment:
This will become a bit nicer with the other PR (that I will continue after this) that has this line as a default string to include, so we only have to type this once :)